src/main/java/org/jboss/logmanager/handlers/WriterHandler.java

```diff
@@ -59,25 +73,66 @@ protected void doPublish(final ExtLogRecord record) {
             return;
         }
         try {
-            lock.lock();
+            write(record, formatted);
+        } catch (Exception ex) {
+            reportError("Error writing log message", ex, ErrorManager.WRITE_FAILURE);
+        }
+    }
+
+    private void lockedWrite(ExtLogRecord record, String formatted) throws IOException {
+        lock.lock();
+        try {
+            doWrite(record, formatted);
+        } finally {
+            lock.unlock();
+        }
+    }
+
+    private boolean doWrite(ExtLogRecord record, String formatted) throws IOException {
```

Comment:
The return value is never used, is that intended?

Reply:
It was like that in the original version do the code but I can change it

Reply:
No need. It won't hurt anything, just noticed it is all :)